tests/gobcore/log/test_gob_logger.py

```diff
@@ -0,0 +1,104 @@
+import mock
+import unittest
+
+from unittest.mock import MagicMock
+
+from tests.gobcore import fixtures
+
+from gobcore.log import logger
+
+
+class TestGobLogger(unittest.TestCase):
+
+    @mock.patch('gobcore.log.GobLogger.info', MagicMock())
+    def testConstructor(self):
+        dummy_msg = fixtures.get_dummy_msg()
+        logger.configure(dummy_msg, "info")
+
+        messages = ["foo", "bar"]
+        for msg in messages:
+            logger.info(msg)
+            logger.info.assert_called_with(msg)
+
+        self.assertEquals(logger.info.call_count, 2)
+
+    @mock.patch('gobcore.log.LOG_PUBLISHER', MagicMock())
+    def testLogLevel(self):
+
+        dummy_msg = fixtures.get_dummy_msg()
+        logger_name = "TEST"
+        loglevel_str = "INFO"
+        expected_messages = []
+
+        logger.configure(dummy_msg, logger_name, 20)
+        with self.assertLogs(logger_name, level=loglevel_str) as result:
+            messages = ["foo", "bar"]
+            for msg in messages:
+                expected_messages.append(fixtures.construct_logging_output(loglevel_str, logger_name, msg))
+                logger.info(msg)
+            self.assertListEqual(result.output, expected_messages)
+
+    @mock.patch('gobcore.log.LOG_PUBLISHER', MagicMock())
+    def testLogLevelChangeLogLevel(self):
+        dummy_msg = fixtures.get_dummy_msg()
+        logger_name = "TEST"
+        logger.configure(dummy_msg, logger_name, 50)
+
+        # if no logging is found, assertError is thrown.
+        with self.assertLogs():
+            logger.configure(dummy_msg, logger_name, 10)
+            messages = ["foo", "bar"]
+            for msg in messages:
+                logger.info(msg)
+
```

Comment:
assertLogs asserts that at least one message is logged.
The test method is called to test change log level. But how is the change of loglevel tested?

Reply:
`with self.assertLogs(logger_name, level=loglevel_str) as result:`
assertLogs set the log level of logger with that name.